File: 7-reverse-integer/reverse-integer.java
class Solution {
    public int reverse(int x) {
        StringBuilder stringBuilder = new StringBuilder();
        if (x < 0){
            stringBuilder.append(Math.abs(x));
            stringBuilder.append("-");
            stringBuilder.reverse();
        }else {
            stringBuilder.append(Math.abs(x));
            stringBuilder.reverse();
        }
        String resultString = stringBuilder.toString();
        try {
            return Integer.parseInt(resultString);
        }catch (NumberFormatException e){
            return 0;
        }
    }
}

File: 7-reverse-integer/reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        if(x == INT_MIN) return 0;
        long ans = 0;
        bool negative = false;
        if (x < 0)
            negative = true;
        x = abs(x);
        while (x > 0) {
            int temp = x % 10;

            ans = ans * 10 + temp;
            if (ans > INT_MAX)
                return 0;
            x /= 10;
        }
        if (negative)
            return -1 * ans;
        else
            return ans;
    }
};